backend/src/main/java/gov/cdc/usds/simplereport/api/pxp/PatientSelfRegistrationController.java

```diff
@@ -112,12 +116,19 @@ public boolean isExistingPatient(
 
   @GetMapping("/entity-name")
   public String getEntityName(
-      @RequestParam String patientRegistrationLink, HttpServletRequest request) {
+      @RequestParam @Pattern(regexp = "(?i)[2346789abcdefghjkmnpqrtuvwxyz]{5}")
```

Comment:
Unfortunately, I'm not sure that this regex is permissive enough for prod either 😞 We have a couple of patient registration links from when the feature was first rolled out that appear to be hand-created (see attached screenshot from prod Metabase). No new custom links should be created, but we don't want to invalidate the existing ones. Maybe we could put these custom links into a list and validate that the pattern either matches this regex OR is part of the approved list?

![Screen Shot 2022-07-28 at 10 52 01 AM](https://user-images.githubusercontent.com/80282552/181604551-f658ce66-7057-466b-8157-a27beb42e0f2.png)

Reply:
I was worried about that 🙁 thanks for checking. Storing these pre-existing links in a list is probably the only sensible way to go - stay tuned.